### DISCUSSION TITLE
Input out-of-bounds values in level editor

### DISCUSSION THREAD

**Discussion by Blue-Beaker** [Bithack/principia#179] [General]
Most number values in principia has boundaries, but some of them can be useful in creating custom levels when set out-of-bounds.
For example, friction of rubber can be set to zero to make a "ice block"; wavelength of laser can be set to super high to insta-kill any robots it aimed at.

In the past these can only be achieved with glitch or hacks, but I suggest to add a setting to "allow" inputting these unbounded number values in the level editor.

For values in sliders, I suggest to add a input box for every slider to input these unbounded values. For sliders in UI, maybe doubleclick to edit the value directly?

**Comment by rollerozxa** [comment]
I'm not against level hacking and am okay with people playing around with values in the level format to make things as crazy as possible (I have done a lot of this myself). However you do have to keep in mind that whatever lays beyond there is complely unpredictable as well as unsupported, so you are mostly on your own. Generally anything you make in the sandbox with the tools and options provided by the game should always work in the future (barring glitches and bugs that cause behavioural changes when fixed, but the level versioning system should alleviate some of that). Exposing arbitrary out-of-bounds values directly in the sandbox would completely break that promise of compatibility. 

As for out-of-bounds values that are completely safe to use, e.g. increasing the condenser max value above what's currently possible (as I did in my 32-bit controllable binary counter level), their limit can simply be increased officially. It would just be a simple adjustment to the slider without requiring a new level version.

**Reply by Blue-Beaker** [reply]
Yes, maybe increasing the limits within safe range is enough. But I think the slider can be kept like before, because the slider will become hard to use if the range is too large. 
Input boxes are better for these safe-to-use big values, like condenser max, zero or positive rubber friction, any positive but not infinite density for plastic and weights, higher power voltage, etc.
